src/identity/identity.service.ts

```diff
@@ -35,33 +38,42 @@ export class IdentityService {
     return deriveAddress({ public: publicKey }, secret, chainIdOf(address));
   }
 
-  async getDerivedIdentity(address: string, secret: string): Promise<object> {
+  async getDerivedIdentity(address: string, secret: string): Promise<DIDDocument> {
     const publicKey = await this.storage.getPublicKey(address);
 
     if (!publicKey) {
       return null;
     }
 
-    return this.asDidDocument(`${address}:derived:${secret}`, address, publicKey);
+    return this.asDidDocument(`${address}:derived:${secret}`, publicKey);
   }
 
-  asDidDocument(id: string, address: string, publicKey: string): object {
-    return {
+  async asDidDocument(id: string, publicKey: string): Promise<DIDDocument> {
+    const verificationMethods = await this.verificationMethodService.getMethodsFor(id);
+    const didDocument: DIDDocument = {
       '@context': 'https://www.w3.org/ns/did/v1',
-      'id': `did:lto:${id}`,
-      'verificationMethod': [{
-        id: `did:lto:${address}#key`,
+      id: `did:lto:${id}`,
+      verificationMethod: [{
+        id: `did:lto:${id}#key`,
         type: 'Ed25519VerificationKey2018',
-        controller: `did:lto:${address}`,
+        controller: `did:lto:${id}`,
         publicKeyBase58: publicKey,
-        blockchainAccountId: `${address}@lto:${chainIdOf(address)}`,
-      }],
-      'authentication': [
-        `did:lto:${address}#key`,
-      ],
-      'assertionMethod': [
-        `did:lto:${address}#key`,
-      ],
+        blockchainAccountId: `${id}@lto:${chainIdOf(id)}`,
+      }]
     };
+
+    for (const verificationMethod of verificationMethods) {
+      const didVerificationMethod = verificationMethod.asDidMethod(publicKey);
+      didDocument.verificationMethod.push(didVerificationMethod);
+      const didReference = [didVerificationMethod.id];
+
+      if (verificationMethod.isAuthentication()) didDocument.authentication = didReference;
+      if (verificationMethod.isAssertionMethod()) didDocument.assertionMethod = didReference;
+      if (verificationMethod.isKeyAgreement()) didDocument.keyAgreement = didReference;
+      if (verificationMethod.isCapabilityInvocation()) didDocument.capabilityInvocation = didReference;
+      if (verificationMethod.isCapabilityDelegation()) didDocument.capabilityDelegation = didReference;
+    }
+
+    return didDocument;
```

Comment:
If there are **no** verification methods through associations, it's an implicit DID. In that case, `did:lto:${address}#key` needs  to be set as `authentication`, `assertionMethod`, and `capabilityInvocation`.

Reply:
done